reframe/core/schedulers/lsf.py

```diff
@@ -30,45 +31,60 @@ def __init__(self):
             f'schedulers/@{self.registered_name}/job_submit_timeout'
         )
 
-    def emit_preamble(self, job):
-        num_tasks_per_node = job.num_tasks_per_node or 1
-        num_nodes = job.num_tasks // num_tasks_per_node
+    def _format_option(self, var, option):
+        if var is not None:
+            return self._prefix + ' ' + option.format(var)
+        else:
+            return ''
 
+    def emit_preamble(self, job):
         preamble = [
-            self._format_option(f'-J {job.name}'),
-            self._format_option(f'-o {job.stdout}'),
-            self._format_option(f'-e {job.stderr}'),
-            self._format_option(f'-nnodes {num_nodes}')
+            self._format_option(job.name, '-J {0}'),
+            self._format_option(job.stdout, '-o {0}'),
+            self._format_option(job.stderr, '-e {0}'),
         ]
 
+        if job.num_tasks_per_node is not None:
+            num_nodes = job.num_tasks // job.num_tasks_per_node
+            preamble.append(self._format_option(num_nodes, '-nnodes {0}'))
+        else:
+            preamble.append(self._format_option(job.num_tasks, '-n {0}'))
+
+        if job.num_cpus_per_task is not None:
+            num_physical_cores = job.num_tasks * job.num_cpus_per_task
```

Comment:
Following up on my other comment at the PR, I think that you cannot make an assumption about the physical cores here. I would just use the `job.num_cpus_per_task` here directly as an argument to `ptile`.

Trying to fix the unit tests I came across with the oversubscription problem that you described. The `fake_job` in unit tests defines this:

```python
    ret.num_tasks = 16
    ret.num_tasks_per_node = 2
    ret.num_tasks_per_core = 1
    ret.num_tasks_per_socket = 1
    ret.num_cpus_per_task = 18
```

And that assumed 18*16 = 288 tasks, which as a writer of a test setting the `num_cpus_per_task = 18` i wouldn't expect.